aiven/client/cli.py

```diff
@@ -3144,43 +3144,54 @@ def service__flink__table__list(self):
-    @arg("-n", "--table-name", required=True, help="Table name")
-    @arg("--kafka-topic", required=False, help="Topic name, used as a source/sink. (Kafka integration only)")
+    @arg("integration_id", default=None, nargs="?", help="Service integration ID (deprecated)")
+    @arg("-n", "--table-name", required=False, help="Table name (deprecated)")
+    @arg("--kafka-topic", required=False, help="Topic name, used as a source/sink. (Kafka integration only) (deprecated)")
     @arg(
         "--kafka-connector-type",
         required=False,
-        help="Kafka connector type (Kafka integration only)",
+        help="Kafka connector type (Kafka integration only) (deprecated)",
         choices=["upsert-kafka", "kafka"]
     )
     @arg(
         "--kafka-key-format",
         required=False,
-        help="Key format. (Kafka integration only)",
+        help="Key format. (Kafka integration only) (deprecated)",
         choices=["avro", "avro-confluent", "debezium-avro-confluent", "debezium-json", "json"]
     )
-    @arg("--kafka-key-fields", nargs="*", default=[], required=False, help="Key fields names used in table schema")
+    @arg(
+        "--kafka-key-fields",
+        nargs="*",
+        default=[],
+        required=False,
+        help="Key fields names used in table schema (Kafka integration only) (deprecated)"
+    )
     @arg(
         "--kafka-value-format",
         required=False,
-        help="Value format. (Kafka integration only)",
+        help="Value format. (Kafka integration only) (deprecated)",
         choices=["avro", "avro-confluent", "debezium-avro-confluent", "debezium-json", "json"]
     )
     @arg(
         "--kafka-startup-mode",
         required=False,
-        help="Kafka startup mode (Kafka integration only)",
+        help="Kafka startup mode (Kafka integration only) (deprecated)",
         choices=["earliest-offset", "latest-offset"]
     )
-    @arg("--jdbc-table", required=False, help="Table name in Database, used as a source/sink. (PG integration only)")
-    @arg("--opensearch-index", required=False, help="OpenSearch index name. (OpenSearch integration only)")
+    @arg(
+        "--jdbc-table",
+        required=False,
+        help="Table name in Database, used as a source/sink. (PG integration only) (deprecated)"
+    )
+    @arg("--opensearch-index", required=False, help="OpenSearch index name. (OpenSearch integration only) (deprecated)")
     @arg(
         "-l",
         "--like-options",
         required=False,
-        help="Clause can be used to create a table based on a definition of an existing table"
+        help="Clause can be used to create a table based on a definition of an existing table (deprecated)"
     )
-    @arg("-s", "--schema-sql", required=True, help="Source/Sink table schema")
+    @arg("-s", "--schema-sql", required=False, help="Source/Sink table schema (deprecated)")
+    @arg.json_path_or_string_parameter("--table-properties")
```

Comment:
Mo hint of the format for the JSON? Should there be - it's pretty complex.

And no client side validation?

Reply:
Nothing apart of valid JSON check.